posting/writer_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package posting
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math"
+	"testing"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/dgraph-io/badger/v2/options"
+	"github.com/dgraph-io/dgraph/x"
+)
+
+type kv struct {
+	key   []byte
+	value []byte
+}
+
+var tmpIndexDir, err = ioutil.TempDir("", "dgraph_index_")
+
+var dbOpts = badger.DefaultOptions(tmpIndexDir).
+	WithSyncWrites(false).
+	WithNumVersionsToKeep(math.MaxInt64).
+	WithLogger(&x.ToGlog{}).
+	WithCompression(options.None).
+	WithEventLogging(false).
+	WithLogRotatesToFlush(10).
+	WithMaxCacheSize(50) // TODO(Aman): Disable cache altogether
+
+var db, err2 = badger.OpenManaged(dbOpts)
+
+func createKVList() []kv {
+	var KVList = []kv{}
+	for i := 0; i < 50000; i++ {
+		n := kv{key: []byte(string(i)), value: []byte("Check Value")}
+		KVList = append(KVList, n)
+	}
+	return KVList
+}
+
+func BenchmarkTxnWriter(b *testing.B) {
+	KVList := createKVList()
+	w := NewTxnWriter(db)
```

Comment:
This writer should be creating in the `for loop`. You would get an error on using this txn writer after calling flush.

Reply:
Great! It works now. What does flush do that I need to recreate the instance? I was under impression it would only flush the contents and not deference the pointer/clear the instance.

Reply:
I was wrong. I thought the error is thrown from this writer but it is thrown by the badger txn. I've commented on the correct line.